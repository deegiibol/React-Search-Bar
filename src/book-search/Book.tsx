import React from 'react';
import { connect } from 'react-redux';
import { addToWishlist } from '../redux/actions/bookActions';
import { IBook } from './Models';

const Book = (props: BookProps) => {
  const addBookToWishlist = () => {
    props.addToWishlist(props);
  };

  return (
    <div className="book" onClick={() => addBookToWishlist()}>
      <div className="thumbnail">
        <img src={props.imageLinks.smallThumbnail} />
      </div>
      <div className="description">
        <h3>{props.title}</h3>
        <h5>
          <span>Author: {props.authors}</span>
        </h5>
        <span>
          Published: {props.publisher} /{props.publishDate}/
        </span>
        <p>{props.description}</p>
      </div>
    </div>
  );
};

interface BookProps extends IBook {
  addToWishlist(book: IBook): void;
}

export default connect(null, { addToWishlist })(Book);
