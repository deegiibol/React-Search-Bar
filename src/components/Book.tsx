import React from 'react';
import { BookProps } from './Models';

const Book = (props: BookProps) => {
  const addBookToWishlist = () => {
    props.addToWishlist(props);
  };

  return (
    <div className="book" onClick={() => addBookToWishlist()}>
      <div className="thumbnail">
        <img src={props.imageLinks.smallThumbnail} alt="Imange not found" />
      </div>
      <div className="description">
        <h3>{props.title}</h3>
        <h5>
          <span>Author: {props.authors}</span>
        </h5>
        <span>
          Published: {props.publisher} /{props.publishDate}/
        </span>
        <p>{props.description}</p>
      </div>
    </div>
  );
};

export default Book;
