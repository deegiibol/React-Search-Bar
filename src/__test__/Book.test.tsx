import React from 'react';
import renderer from 'react-test-renderer';
import Book from '../components/Book';

const mockProp = {
  id: '123',
  title: 'Java',
  authors: ['Tom'],
  publisher: 'Test',
  publishDate: '2015/10/13',
  description: 'Java Core',
  pageCount: 200,
  categories: ['Java'],
  imageLinks: {
    smallThumbnail: '',
    thumbnail: '',
  },
  language: 'eng',
  addToWishlist: jest.fn(() => {}),
};

describe(`The Book Component test`, () => {
  it(`renders as expected`, () => {
    const tree = renderer.create(<Book {...mockProp} />);
    console.log(tree.toJSON());
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
