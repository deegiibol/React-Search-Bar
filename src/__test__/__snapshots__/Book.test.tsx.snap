// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Book Component test renders as expected 1`] = `
<div
  className="book"
  onClick={[Function]}
>
  <div
    className="thumbnail"
  >
    <img
      alt="Imange not found"
      src=""
    />
  </div>
  <div
    className="description"
  >
    <h3>
      Java
    </h3>
    <h5>
      <span>
        Author: 
        Tom
      </span>
    </h5>
    <span>
      Published: 
      Test
       /
      2015/10/13
      /
    </span>
    <p>
      Java Core
    </p>
  </div>
</div>
`;
